adapters/repos/db/vector/hnsw/startup.go

```diff
@@ -117,6 +122,54 @@ func (h *hnsw) restoreFromDisk() error {
 		h.metrics.TrackStartupIndividual(beforeIndividual)
 	}
 
+	if len(fileNames) > 0 {
+		// some commit files has been processed, a new snapshot is created
+		snapshotFileName := snapshotFileName(fileNames[len(fileNames)-1])
+
+		tmpSnapshotFileName := fmt.Sprintf("%s.tmp", snapshotFileName)
+
+		snap, err := os.OpenFile(tmpSnapshotFileName, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0o666)
+		if err != nil {
+			return errors.Wrapf(err, "create snapshot file %q", tmpSnapshotFileName)
+		}
+		defer snap.Close()
+
+		w := bufio.NewWriter(snap)
+
+		err = writeStateTo(state, w)
+		if err != nil {
+			return errors.Wrapf(err, "writing snapshot file %q", tmpSnapshotFileName)
+		}
+
+		err = w.Flush()
+		if err != nil {
+			return errors.Wrapf(err, "flushing snapshot file %q", tmpSnapshotFileName)
+		}
+
+		err = snap.Sync()
+		if err != nil {
+			return errors.Wrapf(err, "fsync snapshot file %q", tmpSnapshotFileName)
+		}
+
+		err = snap.Close()
+		if err != nil {
+			return errors.Wrapf(err, "close snapshot file %q", tmpSnapshotFileName)
+		}
+
+		err = os.Rename(tmpSnapshotFileName, snapshotFileName)
+		if err != nil {
+			return errors.Wrapf(err, "rename snapshot file %q", tmpSnapshotFileName)
+		}
+	}
+
+	// commit log files can be deleted because an snapshot includes its actions
+	for _, fileName := range fileNames {
+		err := os.Remove(fileName)
```

Comment:
Not sure if it was by design, but it seems to be keeping the last two snapshots instead of one.
- the call to `getLastSnapshot` (L.62) makes sure there is only one snapshot left
- then in this condition we create a new snapshot if we found a commit log
- then in this line we remove the commit log: should we remove the previous snapshot as well?

Reply:
it's by design so to be sure if something goes wrong we still have the previous consolidated one. Older snapshots will be erased when restarting...

Reply:
the older snapshot can be removed as soon as the new one is synced on disk